File: tmdb-core/src/main/java/dev/reuss/tmdb/value/id/CreditId.java
package dev.reuss.tmdb.value.id;

/**
 * Strongly typed identifier for a TMDB credit resource.
 *
 * <p>TMDB credit ids are string values.</p>
 *
 * @param value the raw TMDB credit id
 */
public record CreditId(String value) implements TmdbResourceId {

    public CreditId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Credit id must not be blank");
        }

        value = value.trim();
    }

    public static CreditId of(String value) {
        return new CreditId(value);
    }

    @Override
    public String asString() {
        return value;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/value/id/CreditId.kt
package dev.reuss.tmdb.value.id

/**
 * Strongly typed identifier for a TMDB credit resource.
 *
 * TMDB credit ids are string values.
 *
 * @property value the raw TMDB credit id
 */
class CreditId private constructor(
    value: String,
) : StringTmdbResourceId(value, "Credit id") {
    companion object {
        /**
         * Creates an identifier from a non-blank value.
         *
         * Leading and trailing whitespace is removed.
         *
         * @throws IllegalArgumentException if [value] is blank
         */
        @JvmStatic
        fun of(value: String): CreditId = CreditId(value)
    }
}
